### PR TITLE
Add support for deleting a single item from map

### PR DESCRIPTION
**Pull request by stanek-michal**
Also make sure to report errors from bpf_map_get_next_key

### CODE REVIEW

**Review by ferullo — APPROVED** [review]
LGTM if it works.

This will be released in a patch release. Is the merge target ok for that?

**Comment by stanek-michal** [comment]
> This will be released in a patch release. Is the merge target ok for that?

Not really, I was planning on merging this to `main` while actually checking out `delete_item_from_map_oldhead` in Endpoint, which is this single commit on top of the previous checked out Ebpf revision. This is because some additional work is needed to get Endpoint to work with `main` (modify build scripts, test if legacy kprobe loading still works on old kernels, switch to new toolchain in CI).

**Comment by ferullo** [comment]
Hmm. Should we start a 7.16 branch of this repo then?

**Comment by ferullo** [comment]
Thinking it more, we definitely should start a 7.16 branch based off the commit haha 7.16 has been released from in the past.

**Review by fntlnz — APPROVED** [review]
LGTM, thanks @stanek-michal

**Comment by fntlnz** [comment]
Starting a 7.16 branch looks like a great idea, main is also under heavy development right now.

**Review by intxgo — APPROVED** [review]

File: non-GPL/HostIsolationMapsUtil/UpdateMaps.c [hunk 1]
```diff
@@ -197,16 +269,27 @@ ebpf_clear_map(const char *map_path,
...
+        {
+            // map is already empty
+            rv = 0;
+            goto cleanup;
+        }
+        else
+        {
+            // failure (perhaps not supported)
+            rv = -1;
+            ebpf_log("Error getting next key while clearing map, errno=%d\n", errno);
```
Comment (ferullo) [inline non-GPL/HostIsolationMapsUtil/UpdateMaps.c:282]: this line hits for me on a 4.15 kernel with errno 524

Comment (nicholasberlin) [reply non-GPL/HostIsolationMapsUtil/UpdateMaps.c:282]: perhaps we need to clear `errno` before calling `bpf_map_get_next_key`?

**Review by nicholasberlin — APPROVED** [review]

File: non-GPL/HostIsolationMapsUtil/UpdateMaps.c (hunk 1 above)
Comment (stanek-michal) [reply non-GPL/HostIsolationMapsUtil/UpdateMaps.c:282]: I answered in the Endpoint PR which uses this - it turned out lpm_trie maps didn't have support for deleting keys until around 4.17.

**Pull request merged**